sympy/core/compatibility.py

```diff
@@ -763,7 +763,7 @@ def _make_key(args, kwds, typed,
         return key[0]
     return _HashedSeq(key)
 
-def lru_cache(maxsize=100, typed=False):
+def lru_cache(maxsize=1024, typed=False):
```

Comment:
There is no need to change this default. This is a fallback implementation that is only used on Python 2 in case fastcache is not available. This has no effect on Python 3 lru_cache or fastcache. The proper place for setting the new size is the decorator in numbers. py: `@lru_cache(1024)` instead of `@lru_cache()`.